DANS-KNAW/dd-manage-deposit
# DD-1563 Some improvement points to the dd-manage-deposits

Fixes DD-1563 Some improvement points to the dd-manage-deposits 

# Description of changes
  - Avoid exception where possible and replace them with logger Errors
  - Return meaningful Responses to clients (Success-ed of Failed + reason)
  - Check and when need give Error when a deposit.proerties file is directly placed/created in one of the base service folders (see module config file)
    - Unit test this change
  - Update usage 
# How to test
  - Starting vagrant
    - `start-preprovisioned-box.py -s dev_archaeology`
    - `deploy.py -m dd-manage-deposit dev_archaeology`
  
  - Deposit vaild/invalid datasets using
    - cd to local folder: `~/git/dans-core-systems/modules/dd-dans-sword2-examples` 
    - run example: `./run-simple-deposit.sh https://dev.sword2.archaeology.datastations.nl/collection/1 user001 user001 src/main/resources/example-bags/valid/audiences`
    
  - Testing a dataset with multiple deposit.properties files (only one database record should be created)
    -   Unzip attached  '14547798/c15ee79c-d6e3-46d0-aa63-bf8ea23fd310.zip' to local vagrant 'shared' folder
    - In the Terminal open vagrant ssh (`vagrant ssh dev_archaeology`)
    - Copy the unzipped folder to '/var/opt/dans.knaw.nl/tmp/sword2-uploads'. For example: 
      - `cp -r /vagrant/provisioning/shared//c15ee79c-d6e3-46d0-aa63-bf8ea23fd310/ /var/opt/dans.knaw.nl/tmp/sword2-
    - uploads` 
   
  - Looking at the results:
    - in the terminal:
      -    open vagrant ssh (`vagrant ssh dev_archaeology`)
      - `tree -L 5 /var/opt/dans.knaw.nl/tmp` or tree -L 6 /var/opt/dans.knaw.nl/tmp`
    - Reports:
      - in the terminal: `curl 'accept: text/csv' http://dev.sword2.archaeology.datastations.nl:20355/report`  
      - in the firefox browser: `http://dev.sword2.archaeology.datastations.nl:20355/report`
      - in vagrant box: cd /home/dd-manage-deposit
        - Check the user001.csv file 

# Related PRs

(Add links)

*

# Notify

@DANS-KNAW/core-systems
